code/obj/item/assembly/misc_assemblies.dm

```diff
@@ -882,29 +882,43 @@ obj/item/assembly/radio_horn/receive_signal()
 	var/thingsneeded = null
 	var/obj/item/ammo/bullets/result = null
 	var/obj/item/accepteditem = null
+	var/craftname = null
+	var/obj/item/material = null
 
 	proc/craftwith(obj/item/craftingitem, obj/item/frame, mob/user)
+
 		if (istype(craftingitem, accepteditem)) //success! items match
 			src.thingsneeded --
+			if (craftingitem.amount > 1)//it's a little weird but it's one of the lesser messy ways
+				material = craftingitem.split_stack(1)
```

Comment:
Why do you only want to shove one at a time. If the amount is greater than `1`, why not let them shove in `min(src.thingsneeded, craftingitem.amount)`?